arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -944,50 +910,24 @@ public fun <A, B> Iterable<Ior<A, B>>.unalign(): Pair<List<A>, List<B>> = separa
 public inline fun <A, B, C> Iterable<C>.unalign(fa: (C) -> Ior<A, B>): Pair<List<A>, List<B>> =
   map(fa).unalign()
 
-/**
- * Separate the inner [Ior] values into a pair of Lists.
- *
- * @receiver Iterable of Ior
- * @return a tuple containing a List with the left side value from the[Ior.Left] and [Ior.Both] values and another List with the right side value from the [Ior.Right] and [Ior.Both] values.
- */
-public fun <A, B> Iterable<Ior<A, B>>.separateIor(): Pair<List<A>, List<B>> =
-  fold(emptyList<A>() to emptyList()) { (l, r), x ->
-    x.fold(
-      { l + it to r },
-      { l to r + it },
-      { a, b -> l + a to r + b }
-    )
-  }
-
-@Deprecated("Use fold instead", ReplaceWith("fold(MA.empty(), MA::combine)", "arrow.typeclasses.combine"))
+@Deprecated("Use fold from Kotlin Std instead", ReplaceWith("fold(MA.empty(), MA::combine)", "arrow.typeclasses.combine"))
 public fun <A> Iterable<A>.combineAll(MA: Monoid<A>): A =
   fold(MA.empty(), MA::combine)
 
 /**
- * Returns the first element as [Some(element)][Some], or [None] if the iterable is empty.
+ * Returns the first element as [Some], or [None] if the iterable is empty.
  */
 public fun <T> Iterable<T>.firstOrNone(): Option<T> =
   when (this) {
-    is Collection -> if (!isEmpty()) {
-      Some(first())
-    } else {
-      None
-    }
-
-    else -> {
-      iterator().nextOrNone()
-    }
+    is Collection -> if (!isEmpty()) Some(first()) else None
```

Comment:
I'm very surprised by this line, are there cases where doing this is faster?

Reply:
That's a very good question.. This was the original implementation, I just removed some of the braces without thinking too much about it.

The Kotlin Std makes the same optimisations for [`Iterable.firstOrNull`](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/common/src/generated/_Collections.kt#L260) so I guess it's faster. I am not sure about the details though.